redwood/api/src/services/cachedProfiles/cachedProfiles.ts

```diff
@@ -58,10 +62,110 @@ const STATUS_ENUMS: {[enumVal: number]: StatusEnum} = {
 export const parseChallengeStatus = (status: number): StatusEnum =>
   STATUS_ENUMS[status]
 
-export const CachedProfile = {
-  // TODO: duplicate logic from profile.cairo#_get_current_status
-  status: (profile): ProfileStatus => 'NOT_CHALLENGED',
+// Keep in sync with profile.cairo#_get_current_status
+
+export const currentStatus = (
+  profile: Pick<
+    PrismaCachedProfile,
+    | 'lastRecordedStatus'
+    | 'challengeTimestamp'
+    | 'adjudicationTimestamp'
+    | 'appealTimestamp'
+  >,
+  now = new Date()
```

Comment:
in my experience it's easy to make mistakes due to working with raw date objects in javascript. For example it seems hard to verify that e.g. `dayjs(now).diff(profile.challengeTimestamp, 'seconds')` will always give precisely the same answer as `Date.now() - profile.challengeTimestamp`

Would it work to work directly in unix time (seconds or millis) via e.g. `Date.now()`?

Reply:
I agree that doing math with date objects directly is full of footguns and super scary. However, I found that using a good wrapper library like dayjs helps you do the right thing and get a bunch of edge cases taken care of for free.

In this specific circumstance though I think you're right—just using `Date.now()` and doing the math in milliseconds would be better, since the goal here is to faithfully replicate the logic from the Cairo side which is all necessarily done that way.